Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MemoryUseTracker.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Diagnostics;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.DiagnosticsSystem
+{
+    internal class MemoryUseTracker
+    {
+        private Process currentProcess = Process.GetCurrentProcess();
+        private MemoryReading[] memoryReadings = new MemoryReading[10];
+        private int index = 0;
+
+        public MemoryReading GetReading()
+        {
+            var reading = new MemoryReading()
```

Comment:
Instead of making a new one each time, and we know we always have 10 in our array, why not just reuse the ones we have if we've already filled the array?